drools-wb-screens/drools-wb-test-scenario-editor/drools-wb-test-scenario-editor-client/src/main/java/org/drools/workbench/screens/testscenario/client/VerifyFactsPanel.java

```diff
@@ -61,27 +62,46 @@ public VerifyFactsPanel( FixtureList verifyFacts,
                 column.add( new DeleteButton( verifyFact ) );
 
                 add( column );
+            } else if (fixture instanceof VerifyScorecardScore) {
+                VerifyScorecardScore verifyScorecardScore = (VerifyScorecardScore) fixture;
+
+                HorizontalPanel column = new HorizontalPanel();
+                column.add(new VerifyScorecardScoreWidget(verifyScorecardScore,
+                                                          showResults));
+
+                column.add(new DeleteButton(verifyScorecardScore));
+
+                add(column);
             }
         }
     }
 
     class DeleteButton extends Button {
 
-        public DeleteButton( final VerifyFact verifyFact ) {
+        public DeleteButton( final Fixture fixture ) {
             setIcon(IconType.TRASH);
-            setText("'" + verifyFact.getName() + "'");
+            setText("'" + getName(fixture) + "'");
             setTitle(TestScenarioConstants.INSTANCE.DeleteTheExpectationForThisFact());
 
             addClickHandler( new ClickHandler() {
 
                 public void onClick( ClickEvent event ) {
                     if ( Window.confirm( TestScenarioConstants.INSTANCE.AreYouSureYouWantToRemoveThisExpectation() ) ) {
-                        scenario.removeFixture( verifyFact );
+                        scenario.removeFixture( fixture );
                         parent.renderEditor();
                     }
                 }
             } );
         }
-    }
 
+        private Object getName(Fixture fixture) {
+            if (fixture instanceof VerifyFact) {
+                return ((VerifyFact) fixture).getName();
+            } else if (fixture instanceof VerifyScorecardScore) {
+                return TestScenarioConstants.INSTANCE.ScoreCardScore();
+            } else {
+                return "";
```

Comment:
This branch has been added by this PR, before that the code was `setText("'" + verifyFact.getName() + "'");`. Are you sure about that? If this should not happen I would prefer to throw an exception as `else`. Wdyt?

Reply:
Yes here exception makes sense. This should always be either of the two types that are in the top if clauses.